ext/mongo_kerberos/mongo_kerberos_native.c

```diff
@@ -115,7 +128,9 @@ static VALUE initialize_challenge(VALUE self) {
     return Qfalse;
   }
 
-  result = sasl_encode64(raw_payload, raw_payload_len, encoded_payload, sizeof(encoded_payload), &encoded_payload_len);
+  /* I can't tell if the buffer size expected by sasl_encode64 includes */
+  /* the null terminator, thus be defensive and exclude it */
+  result = sasl_encode64(raw_payload, raw_payload_len, encoded_payload, sizeof(encoded_payload)-1, &encoded_payload_len);
```

Comment:
Looks to me like it's documented as null-terminated: https://docs.oracle.com/cd/E86824_01/html/E54774/sasl-encode64-3sasl.html